mdsort.c

```diff
@@ -70,6 +74,12 @@ main(int argc, char *argv[])
 		case 'v':
 			log_level++;
 			break;
+		case 'D':
+			if ((eq = strchr(optarg, '=')) == NULL)
+				err(1, "invalid macro: %s", optarg);
+			*eq = '\0';
+			macros_insertc(&macros, optarg, eq + 1);
```

Comment:
`macros_insertc` must only be used when the number of macros are known at compile time, see the comment attached to the implementation:

```
$ mdsort -D a=a -D b=b -D c=c        
mdsort: macros_insertc: stack storage exhausted
```

This routine allows macros to be used without requiring any heap allocation. Instead, `macros_insert` must be used here.